### ISSUE TITLE
Experimental options (async/await) not working with NodeCompiler

### ISSUE THREAD

**Issue by hendrikswan** [google/traceur-compiler#1711]
I'm trying to use async/await with gulp-traceur. I'm passing in the experimental option. 

The following code produces an error `Error filename:line:char: Semi-colon expected`, but compiles correctly when I try it out in the traceur transcoding demo ([see for yourself](http://google.github.io/traceur-compiler/demo/repl.html#%2F%2F%20Options%3A%20--annotations%20--array-comprehension%20--async-functions%20--exponentiation%20--generator-comprehension%20--member-variables%20--require%20--symbols%20--types%20%0Afunction%20promisingOperation()%20%7B%0A%20%20%20%20return%20new%20Promise(function(resolve%2C%20reject)%20%7B%0A%20%20%20%20%20%20%20%20setTimeout(function()%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20if(%20Math.round(Math.random())%20)%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20resolve('Success!')%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20else%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20reject('Failure!')%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D%2C%201000)%3B%0A%20%20%20%20%7D)%3B%0A%7D%0A%0Aasync%20function%20test()%20%7B%0A%20%20%20%20var%20message%20%3D%20await%20promisingOperation()%3B%0A%20%20%20%20console.log(message)%3B%0A%7D%3B%20%0A%0A%0A)): 

```javascript
function promisingOperation() {
    return new Promise(function(resolve, reject) {
        setTimeout(function() {
            if( Math.round(Math.random()) ){
                resolve('Success!');
            }
            else{
                reject('Failure!');
            }
        }, 1000);
    });
}

(async function test() {
    var message = await promisingOperation();
    console.log(message);
})(); 
```

Have a look at the issue which I logged at gulp-traceur for more context.@sindresorhus replied and said that gulp-traceur is simply passing through the options to the node compiler, and that I should log the issue on this repo.  
https://github.com/sindresorhus/gulp-traceur/issues/60

I also validated that @sindresorhus is correct by doing a little spike where I try to use the NodeCompiler to compile experimental async/await code with the latest release of traceur. Have a look at this little spike repo https://github.com/hendrikswan/node-traceur-spike, it's a clean example of the issue without gulp-traceur.

**Comment by hendrikswan** [comment]
Reading over the issue, I realise that it's saying way too much. For anyone interested in this issue, just have a look at this spike where it illustrates the issue at hand: https://github.com/hendrikswan/node-traceur-spike

**Comment by johnjbarton** [comment]
Try 
```js
var compiler = new traceur.NodeCompiler(
    traceur.Options.experimental().setFromObject({modules: 'commonjs'})
);
```

**Comment by hendrikswan** [comment]
When I try that, I get:
```bash
traceur.Options.experimental().setFromObject({modules: 'commonjs'})
TypeError: Cannot read property 'experimental' of undefined
```

**Comment by johnjbarton** [comment]
Sorry, it's `traceur.util.Options`

**Comment by hendrikswan** [comment]
Ah, seems to be working now! John, do you have any recommendations on how I
could possibly fix this in gulp-traceur? If it's necessary to do it this
way, then obviously just passing through options from the gulp task to
traceur won't work. Should I just if statement the crap out of it? ;)

**Comment by arv** [comment]
@johnjbarton We should make `experimental: true` work as well.

It seems doable...

**Comment by hendrikswan** [comment]
A bit more background. I hacked the traceur.js file in my node_modules folder, trying to make sense of what's going on. At lin 2954, I added the following line:
`console.log('setting experimental value ' + name + ', value ' +  v);`

When I was passing through `experimental: true` to the NodeCompiler, this was the output:

```bash
setting experimental value annotations, value true
setting experimental value arrayComprehension, value true
setting experimental value asyncFunctions, value true
setting experimental value exponentiation, value true
setting experimental value generatorComprehension, value true
setting experimental value require, value true
setting experimental value symbols, value true
setting experimental value types, value true
setting experimental value memberVariables, value true
```

I don't know if that helps, but as far as I can tell the option is making it that far :)

**Comment by hendrikswan** [comment]
If you could point me in the right direction, I could also make the change.

**Comment by johnjbarton** [comment]
As you see from your output, 'experimental' is not an option but a command to the Options class to set various options to a value given.  Consequently the result of using `experimental: true` would depend upon the order of options setting.

**Comment by johnjbarton** [comment]
In addition, the meaning of `experimental === false` is different from the result one might get with `experimental = false`. Anyway more information on https://github.com/google/traceur-compiler/pull/1491

**Comment by UltCombo** [comment]
@johnjbarton Is `experimental` still a method of the Options class? With Traceur 0.0.86, trying to call `new traceur.util.Options().experimental()` throws `TypeError: boolean is not a function`

**Comment by UltCombo** [comment]
Looks like experimental is just a setter again now. https://github.com/google/traceur-compiler/blob/master/src/Options.js#L239-L244

**Comment by UltCombo** [comment]
Oh wait, there's both a setter and a class static method named `experimental`. So, instances have a setter and the class has a static method with the same name, that's where I was getting lost. The correct usage is:

```js
traceur.util.Options.experimental().setFromObject({ modules: 'commonjs' })
```

**Comment by UltCombo** [comment]
Guess I should have paid more attention to JJB's [example above](https://github.com/google/traceur-compiler/issues/1711#issuecomment-73526095). Cleaning up the comments which don't add anything meaningful.

**Comment by guysegal** [comment]
gulp.task('default', function () {
    return gulp.src('src/app.js')
        .pipe(traceur({asyncFunctions: true}))
        .pipe(gulp.dest('dist'));
});

Wored for me.

**Comment by johnjbarton** [comment]
yes setting individual options should work fine.  Only setting experimental fails.
